codec-sockjs/src/main/java/io/netty/handler/codec/sockjs/util/JsonConverter.java

```diff
@@ -0,0 +1,269 @@
+                        buffer.writeByte('\\').writeByte('u');
+                        for (int k = 0; k < 4 - hex.length(); k++) {
+                            buffer.writeByte('0');
+                        }
+                        buffer.writeBytes(hex.toLowerCase().getBytes());
+                    } else {
+                        buffer.writeByte(ch);
+                    }
+            }
+        }
+        return buffer;
+    }
+
+    public static boolean isControlCharacter(final char c) {
+        return c >= '\u0000' && c <= '\u001F';
+    }
+
+    public static boolean isSurrogateCharacter(final char c) {
+        return c >= Character.MIN_HIGH_SURROGATE && c <= Character.MAX_LOW_SURROGATE;
+    }
+
+    public static boolean isFormatControlCharacter(final char c) {
+        return c == '\u200C' || c == '\u200D';
+    }
+
+    public static boolean isSeparatorCharacter(final char c) {
+        return c >= '\u2028' && c <= '\u202F';
+    }
+
+    public static boolean isFormatOtherCharacter(final char c) {
+        return c >= '\u2060' && c <= '\u206F';
+    }
+
+    public static boolean isSpecials(final char c) {
+        return c >= '\uFFF0' && c <= '\uFFFF';
+    }
+
+    /**
+     * Processes the input char[] array and escapes the any control characters, quotes, slashes,
+     * and unicode characters.
+     *
+     * @param value the char[] for which unicode characters should be escaped
+     * @return {@code String} Java style escaped unicode characters.
+     */
+    public static String escapeCharacters(final char[] value) {
+        final StringBuilder buffer = new StringBuilder(value.length);
+        for (char ch : value) {
+            if (shouldEscape(ch)) {
+                final String hex = Integer.toHexString(ch);
+                buffer.append('\\').append('u');
+                for (int k = 0; k < 4 - hex.length(); k++) {
+                    buffer.append('0');
+                }
+                buffer.append(hex.toLowerCase());
+            } else {
+                buffer.append(ch);
+            }
+        }
+        return buffer.toString();
+    }
+
+    private static boolean shouldEscape(final char c) {
+        return isControlCharacter(c) ||
+                isSurrogateCharacter(c) ||
+                isFormatControlCharacter(c) ||
+                isSeparatorCharacter(c) ||
+                isSpecials(c);
+    }
+
+    /**
+     * Taken from
+     * <a href="https://github.com/FasterXML/jackson-docs/wiki/JacksonSampleQuoteChars">JacksonSampleQuoteChars</a>
+     * with a few minor modifictions.
+     */
+    private static class StringSerializer extends JsonSerializer<String> {
+
+        @Override
+        public void serialize(final String str, final JsonGenerator gen, final SerializerProvider provider)
+        throws IOException {
+            final int status = ((JsonWriteContext) gen.getOutputContext()).writeValue();
+            switch (status) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add a default case to this switch. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3ASwitchLastCaseIsDefaultCheck)